ext/sodium/libsodium.c

```diff
@@ -3220,12 +3101,8 @@ PHP_FUNCTION(sodium_pad)
 		sodium_remove_param_values_from_backtrace(EG(exception));
 		RETURN_THROWS();
 	}
-	if (blocksize <= 0) {
-		zend_throw_exception(sodium_exception_ce, "block size cannot be less than 1", 0);
-		RETURN_THROWS();
-	}
-	if (blocksize > SIZE_MAX) {
-		zend_throw_exception(sodium_exception_ce, "block size is too large", 0);
+	if (blocksize <= 0 || blocksize > SIZE_MAX) {
+		zend_argument_error(sodium_exception_ce, 2, "to be greater than 0");
```

Comment:
As 0 is a particular common case: This can also be `to be positive`. Or `to be non-negative` for the `>=` case.

Reply:
I'd prefer to get rid of positive/negative from messages because I feel that the exact number is easier to understand - but maybe it's just me. :)

Furthermore, as far as I remember the position of 0 is not clear everywhere in the world. I found a comment about it: https://math.stackexchange.com/questions/26705/is-zero-positive-or-negative. But I might be wrong here as well.